### DISCUSSION TITLE
Using CliWrap in a RDP Host does not work on some machines

### DISCUSSION THREAD

**Discussion by byrialsen** [Tyrrrz/CliWrap#149] [Q&A]
Hi @Tyrrrz 

I have a C# application that uses Windows ```query session``` command to check for an active RDP session (meaning that the PC is  running as RDP host). 

I started using standard .Net Process class to run the command but I experienced problems on some PC's with ```WaitForExit()```. Running locally it works all the time, but running on some PC's in a RDP host session then ```WaitForExit()``` times out after 30s and StandardOutput also returned after the 30s timeout. I tried a million solutions from the internet, but had no success with any of them.

Then I found CliWrap and it looks like you have put a lot of effort into it 💪.....but unfortunately it seems like the problem with RDP session also exist here. I used code like this:

```csharp
var result =
  await CliWrap.Cli.Wrap(Path.Combine(Environment.SystemDirectory, "query.exe"))
    .WithArguments("session")
    .WithValidation(CliWrap.CommandResultValidation.None)
    .ExecuteBufferedAsync();
```

So my question is. Do you have any clue what can course this behavior and maybe you want to have a fix for this in your solution 😜

**Comment by Tyrrrz** [comment]
Hi @byrialsen 

I have never heard of this issue. Unfortunately CliWrap is built on top of the standard Process class so it inherits all the issues associates with it (but works around them when possible and provides a better API).

If your problem is reproducible with the Process class, I'd highly suggest reporting it to the dotnet team, as it might indicate a bug in .NET, or possibly a compatibility issue with Windows/RDP.
